bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/wizard/NewComponentWizardPage.java

```diff
@@ -112,11 +112,20 @@ public void modifyText(ModifyEvent e)
 		projectNameLbl.setText(Messages.NewComponentWizardPage_ProjectNameLbl);
 		projectCombo = new Combo(container, SWT.BORDER | SWT.READ_ONLY);
 		Optional<IProject> optProject = Optional.ofNullable(EclipseUtil.getSelectedProjectInExplorer());
-		optProject.ifPresent(project -> projectCombo.setText(project.getName()));
 		IProject[] projects = ResourcesPlugin.getWorkspace().getRoot().getProjects();
+		int index = 0;
 		for (IProject project : projects)
 		{
 			projectCombo.add(project.getName());
+			if (optProject.isPresent() && optProject.get().equals(project))
+			{
+				projectCombo.select(index);
+			}
+			else
+			{
+				index += 1;
+			}
+
```

Comment:
Your old approach should still work without this even if it's ready_only. Do you see any problem with that?

`optProject.ifPresent(project -> projectCombo.setText(project.getName()));`


> /**
>  * Sets the contents of the receiver's text field to the
>  * given string.
>  * <p>
>  * This call is ignored when the receiver is read only and
>  * the given string is not in the receiver's list.
>  * </p>
>  * <p>